glean-core/src/storage/mod.rs

```diff
@@ -132,9 +132,9 @@ impl StorageManager {
     ) -> Option<Metric> {
         let mut snapshot: Option<Metric> = None;
 
-        let mut snapshotter = |metric_name: &[u8], metric: &Metric| {
-            let metric_name = String::from_utf8_lossy(metric_name).into_owned();
-            if metric_name == metric_id {
+        let mut snapshotter = |id: &[u8], metric: &Metric| {
+            let id = String::from_utf8_lossy(id).into_owned();
```

Comment:
I was afraid that this would shadow the function parameter with the same name, so, considering that this is contained within the function, I opted for `id` here instead of `metric_id`.